### ISSUE TITLE
[HOLD for payment 2023-12-28] [HOLD for payment regression] [$500] Conversation - In group details page, plus options are shown

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#29379]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.81-4

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Tap on a group chat report
3. Tap plus icon and quickly tap the header

## Expected Result:
In group details page, plus options like split bill, assign task and add attachment must not be shown

## Actual Result:
In group details page, plus options like split bill, assign task and add attachment are shown

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

 

https://github.com/Expensify/App/assets/78819774/bdf405f6-eef4-4d73-a7eb-991730984177

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016354d05fb227f48d</li>
        <li>Upwork Job ID: 1714386765019463680</li>
        <li>Last Price Increase: 2023-10-24</li>
<li>Automatic offers: </li>
<ul>
<li>ishpaul777 | Contributor | 27498473</li>
</ul></li>
    </ul>
</details>

**Comment by JmillsExpensify** [comment]
Can we get a video for this one?

**Comment by JmillsExpensify** [comment]
Regardless, triaging for review and more proposals.

**Comment by lanitochka17** [comment]
@JmillsExpensify Hello

The video is in the  Screenshots/Videos tab

![image](https://github.com/Expensify/App/assets/78819774/5f9d72eb-1e3e-468a-b7d4-9c2acc195474)

**Comment by keisyrzk** [comment]
I checked this with the latest main and I experience two mentioned options presence on all platforms not only those mentioned in the description.

**Comment by keisyrzk** [comment]
Hello!
I'm Krzysztof from Callstack - an expert contributor group and I would like to work on this one.

## Proposal
# Please re-state the problem that we are trying to solve in this issue.
In group details page, plus-click options: `split bill` (this one is already hidden), `assign task` and `add attachment` are shown but should not.

# What is the root cause of that problem?
The described list options are presented.

# What changes do you think we should make in order to solve the problem?
According to the task description - removing thlisted options from the list on "+" button list popup in group chat.

# What alternative solutions did you explore? (Optional)
NA

**Comment by keisyrzk** [comment]
Hello! I'm Krzysztof from Callstack - an expert contributor group and I would like to work on this one.

**Comment by JmillsExpensify** [comment]
> Hello! I'm Krzysztof from Callstack - an expert contributor group and I would like to work on this one.

I went ahead and assigned you, thanks!

**Comment by keisyrzk** [comment]
### Proposal
### Please re-state the problem that we are trying to solve in this issue.
In group details page, plus-click options: split bill (this one is already hidden), assign task and add attachment are shown but should not.
### What is the root cause of that problem?
The described list options are presented.
### What changes do you think we should make in order to solve the problem?
The changes are about to be applied in AttachmentPickerWithMenuItems.js file by conditional set the appearance for the listed options.
### Check if this is a group chat (more than 2 members of the room)
`const isGroupChat = report.participantAccountIDs.length > 2;`
For the “const menuItems” do conditional updates using “isGroupChat”

<img width="447" alt="Zrzut ekranu 2023-10-26 o 09 31 06" src="https://github.com/Expensify/App/assets/18525360/fab5a65b-57c1-4539-913c-66f401c35635">


### What alternative solutions did you explore? (Optional)
NA

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by eVoloshchak** [comment]
I think @ishpaul777's [proposal](https://github.com/Expensify/App/issues/29379#issuecomment-1764756030) is the approach we want to take

1. The user taps the "+" button
2. User immediately taps the chat header
3. The app registers "+" button tap, opens the popup menu
4. Navigation actions (from step 2) is performed, navigating user to chat details page (while the popup menu is still open)

Looks like the issue won't be resolved by conditionally setting the appearance for the listed options (as [proposed](https://github.com/Expensify/App/issues/29379#issuecomment-1780610428) by @keisyrzk), it's caused by a poor performance, with taps registered with a delay. 
However, I'm unable to reproduce this. We recently merged a couple of PR's that improve performance, I they _might_ have resolved this.
@ishpaul777, @keisyrzk, could you verify this is still reproducible? Thanks!

**Comment by JmillsExpensify** [comment]
@eVoloshchak Thoughts on the proposals so far?

**Comment by eVoloshchak** [comment]
> I can still reproduce on Ios native app 

Was able to reproduce this on iOS native app too, thank you

In that case, I think we should proceed with @ishpaul777's [proposal](https://github.com/Expensify/App/issues/29379#issuecomment-1764756030) 
🎀👀🎀 C+ reviewed!

**Comment by NikkiWines** [comment]
Yep, agreed that @ishpaul777's proposal looks good 👍

**Comment by situchan** [comment]
The [PR](https://github.com/Expensify/App/pull/30864) was [reverted](https://github.com/Expensify/App/pull/31457) due to [regression](https://github.com/Expensify/App/issues/31426)

**Comment by JmillsExpensify** [comment]
Alright, thanks! We'll wait for that regression to be fixed and I'll make sure that is reflected in payment summary down the line.

**Comment by JmillsExpensify** [comment]
Still working on follow-up PR

**Comment by JmillsExpensify** [comment]
Still waiting on regression fix.

**Comment by JmillsExpensify** [comment]
@eVoloshchak Mind completing the BZ checklist?

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: there isn't a PR that caused this, this was caused by a race condition and reproducible only as an edge case
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: I don't think an additional discussion is needed, this is an edge case and the bug does not prevent users from using the app


**Regression Test Proposal**

1. Open any chat and upload a file bigger than 24 Mb. Dissmiss the popup error.
2. Tap on the report header, verify the keyboard doesn't pop up
3. Tap on the report header and quickly tap on plus icon, verify popover closes as soon user is navigated to details page
4. Tap on the report header and quickly tap on emoji popover, verify popover closes as soon user is navigated to details page
5. Tap the plus icon and quickly tap the header, verify popover closes as soon user is navigated to details page

**Do we agree 👍 or 👎**

**Comment by JmillsExpensify** [comment]
Thank you! Payment summary: 

 - Issue reporter: QA
 - Contributor: $500 @ishpaul777 
 - Contributor+: $500 @eVoloshchak

**Comment by JmillsExpensify** [comment]
Contributor paid out and @eVoloshchak is paid via Expensify. Closing this issue!

**Comment by eVoloshchak** [comment]
@JmillsExpensify, there was a regression, shouldn't the payment be $250?

**Comment by JmillsExpensify** [comment]
Oh thank you for your honesty! I'll update the payment summary above and request a refund on the contributor payment.

**Comment by JmillsExpensify** [comment]
$250 payment approved for @eVoloshchak based on summary above.
